### PR TITLE
chore: machine fixes, features, and refactors

### PR DESCRIPTION
**Pull request by marcus8448**
Fancy new Circuit Fabricator GUI:

https://user-images.githubusercontent.com/28928887/109746952-22d53200-7b8b-11eb-969b-0610725144af.mp4

* fixes a bug where machines would not load from a save
* fixes the circuit fabricator's slot alignment
* changes `Constants` and subclasses to `interface`s for less `public static final` spam (plus you can technically implement it for easy access to constants)
* fixes some method parameter names (stack -> matrices, v -> delta)
* fixes a bug where item attribute data was never saved and loaded from a stack
* fixes a bug where a dedicated server would crash on startup
* fixed the (packet) deserialization of oxygen in a chunk
* fixes opening machine guis on dedicated servers
* fixes a bug where you couldn't join a dedicated server without crashing

### CODE REVIEW

File: src/main/java/com/hrznstudio/galacticraft/client/gui/screen/ingame/ElectricCompressorScreen.java [hunk 1]
```diff
@@ -67,40 +68,40 @@ protected String getBackgroundLocation() {
...
+        DrawableUtils.drawCenteredString(matrices, textRenderer, getContainerDisplayName(), (this.width / 2), this.y + 6, Formatting.DARK_GRAY.getColorValue());
+        this.drawMouseoverTooltip(matrices, mouseX, mouseY);
     }
 
-    protected void drawCraftProgressBar(MatrixStack stack) {
+    protected void drawCraftProgressBar(MatrixStack matrices) {
         float progress = this.handler.machine.getProgress();
         float maxProgress = this.handler.machine.getMaxProgress();
         float progressScale = (progress / maxProgress);
         // Progress confirmed to be working properly, below code is the problem.
```
Comment (JoeZwet) [inline src/main/java/com/hrznstudio/galacticraft/client/gui/screen/ingame/ElectricCompressorScreen.java:97]: is this still a relevant comment?

**Review by JoeZwet — APPROVED** [review]
:+1:

**Review by JoeZwet — APPROVED** [review]
:+1:

**Comment by EzerArch** [comment]
:+1:

**Pull request merged**